engine.py

```diff
@@ -45,22 +45,24 @@ class BdfCallbacks:
     def __init__(self, registry):
         self.registry = registry
 
-    def check(self, name, property_name):
+    def check(self, name, property_name="condition_met"):
         obj = self.registry.get_instance(name)
         obj.activate()
+        print("checking " + name + ": " + str(obj.is_true(property_name)))  # debug
         return obj.is_true(property_name)
 
     def activate(self, name):
         obj = self.registry.get_instance(name)
         obj.activate()
+        print("activating " + name)  # debug
 
     def register(self, name, object_type):
         self.registry.add(name, object_type)
         return name
 
     def prepare(self, name):
         obj = self.registry.get_instance(name)
-        obj.activate()
+        obj.prepare()
```

Comment:
Wow, that is a bug, everytime we find a bug we must write a unit test for it.

Reply:
You may want to change Engine constructor to accept a custom registry so that you can inject dependency to unit test this function.